recipes/atk/all/conanfile.py

```diff
@@ -1,102 +1,130 @@
+        if not self.conf.get("tools.gnu:pkg_config", check_type=str):
+            self.tool_requires("pkgconf/1.9.3")
+        if hasattr(self, "settings_build") and cross_building(self):
+            self.tool_requires("glib/2.75.2")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        env = VirtualBuildEnv(self)
+        env.generate()
+        if not cross_building(self):
+            env = VirtualRunEnv(self)
+            env.generate(scope="build")
+        tc = MesonToolchain(self)
+        tc.project_options["introspection"] = False
+        tc.project_options["docs"] = False
+        tc.project_options["localedir"] = os.path.join(self.package_folder, "bin", "share", "locale")
+        tc.generate()
+        deps = PkgConfigDeps(self)
+        deps.generate()
 
     def _patch_sources(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
+        apply_conandata_patches(self)
+        replace_in_file(self, os.path.join(self.source_folder, "meson.build"), "subdir('tests')", "#subdir('tests')")
 
     def build(self):
         self._patch_sources()
-        tools.replace_in_file(os.path.join(self._source_subfolder, 'meson.build'),
-            "subdir('tests')",
-            "#subdir('tests')")
-        meson = self._configure_meson()
+        meson = Meson(self)
+        meson.configure()
         meson.build()
 
     def package(self):
-        self.copy(pattern="COPYING", dst="licenses", src=self._source_subfolder)
-        meson = self._configure_meson()
+        copy(self, "COPYING", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        meson = Meson(self)
         meson.install()
-        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
-        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
-        if self.settings.compiler == "Visual Studio":
-            for pdb in glob.glob(os.path.join(self.package_folder, "bin", "*.pdb")):
-                os.unlink(pdb)
-            if not self.options.shared:
-                os.rename(os.path.join(self.package_folder, 'lib', 'libatk-1.0.a'), os.path.join(self.package_folder, 'lib', 'atk-1.0.lib'))
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rm(self, "*.pdb", os.path.join(self.package_folder, "bin"))
+        fix_apple_shared_install_name(self)
+        fix_msvc_libname(self)
 
     def package_info(self):
-        self.cpp_info.libs = tools.collect_libs(self)
-        self.cpp_info.includedirs = ['include/atk-1.0']
-
-    def package_id(self):
-        self.info.requires["glib"].full_package_mode()
+        self.cpp_info.set_property("pkg_config_name", "atk")
+        self.cpp_info.libs = ["atk-1.0"]
+        self.cpp_info.includedirs = [os.path.join("include", "atk-1.0")]
+
+def fix_msvc_libname(conanfile, remove_lib_prefix=True):
```

Comment:
Why did you put this function outside of the recipe class?

Reply:
Because I hope it will become a conan function: https://github.com/conan-io/conan/issues/11866